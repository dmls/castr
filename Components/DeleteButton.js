import React from 'react';
import { View, Text, TouchableOpacity, Alert } from 'react-native';
import { styles } from '../assets/styles/Styles';
import { deleteCollections, deleteCharacter } from '../Storage/Storage';

export const DeleteButton = ({ type, record, navigation, textOnly }) => {
  const handlePress = (callback) => {
    Alert.alert(
      'Confirm Deletion',
      `Are you sure you want to delete ${record.name}?`,
      [
        { text: 'Cancel' },
        { text: 'Delete', onPress: async () => {
          const del = await callback([records]);
          navigation.push('Collections');
        } },
      ]
    );
  };

  const callback = {
    'collections': deleteCollections(),
    'characters': deleteCharacters(),
  }[type];

  return (
    <TouchableOpacity
      style={textOnly ? null : [styles.button, styles.buttonDanger]}
      activeOpacity={0.5}
      onPress={handlePress(callback)}
    >
      <Text style={styles.buttonText}>Delete</Text>
    </TouchableOpacity>
  );
}
