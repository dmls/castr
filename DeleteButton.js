import React from 'react';
import { View, Text, TouchableOpacity, Alert } from 'react-native';
import { styles } from '../assets/styles/Styles';
import { deleteCollections } from '../Storage/Storage';

export const DeleteButton = ({ record, navigation, textOnly }) => {
  const handlePress = () => {
    Alert.alert(
      'Confirm Deletion',
      `Are you sure you want to delete ${record.name}?`,
      [
        { text: 'Cancel' },
        { text: 'Delete', onPress: async () => {
          const del = await deleteCollections([record.id]);
          navigation.push('Collections');
        } },
      ]
    );
  };

  return (
    <TouchableOpacity
      style={textOnly ? null : [styles.button, styles.buttonDanger]}
      activeOpacity={0.5}
      onPress={handlePress}
    >
      <Text style={styles.buttonText}>Delete</Text>
    </TouchableOpacity>
  );
}
