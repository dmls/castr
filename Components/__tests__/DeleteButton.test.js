import React from 'react';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native'; // Import NavigationContainer
import DeleteButton from '../DeleteButton';

const callback= () => { /* callback code */ };
const label = 'LABEL';
const navigate = 'NAV_TO';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('DeleteButton', () => {
  test('renders correctly with default settings', () => {
    const component = renderer.create(
      <NavigationContainer>
        <DeleteButton
          callback={callback}
          label={label}
          navigate={navigate}
        />
      </NavigationContainer>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with textOnly = false', () => {
    const component = renderer.create(
      <NavigationContainer>
        <DeleteButton
          callback={callback}
          label={label}
          navigate={navigate}
          textOnly={false}
        />
      </NavigationContainer>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with textOnly = true', () => {
    const component = renderer.create(
      <NavigationContainer>
        <DeleteButton
          callback={callback}
          label={label}
          navigate={navigate}
          textOnly={true}
        />
      </NavigationContainer>
    );
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
