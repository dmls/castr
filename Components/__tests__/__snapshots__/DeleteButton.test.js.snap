// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton renders correctly with default settings 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#dc3545",
      "borderColor": "#0d6efd",
      "borderRadius": 30,
      "borderWidth": 0,
      "flex": 1,
      "height": 40,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    Delete
  </Text>
</View>
`;

exports[`DeleteButton renders correctly with textOnly = false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#dc3545",
      "borderColor": "#0d6efd",
      "borderRadius": 30,
      "borderWidth": 0,
      "flex": 1,
      "height": 40,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    Delete
  </Text>
</View>
`;

exports[`DeleteButton renders correctly with textOnly = true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#dc3545",
      }
    }
  >
    Delete
  </Text>
</View>
`;
